[Kitty-001] iP

Kitty-001 opened the pull request:
## Kitty~

Compare to the previous version Duke, I:
- add more interactions
- change the personality of the ChatBot

What can you do with Kitty?
1. add tasks, from Todos, Deadlines to Events
2. list out tasks
3. find tasks with specific keyword
4. delete tasks

Feel free to try and give some **FEEDBACK**! 😄 ❤️ 

What to do ~~in the future~~ *this week*???
- [ ] review peer PRs
- [ ] add GUI

### **Want to help???**
I am [here](https://github.com/Kitty-001), feel free to contact me!

> CS2101 and CS2103T provided by NUS is something not to be missed!!!!

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Kitty-001 committed:
Level-0

Kitty-001 committed:
Level-1

Kitty-001 committed:
Level-2

Kitty-001 committed:
Level-3

Kitty-001 committed:
Level-4

Kitty-001 committed:
Level-5

Kitty-001 committed:
Level-6

Kitty-001 committed:
Level-7

Kitty-001 committed:
Merge branch 'branch-Level-7'

Kitty-001 committed:
Level-8

Kitty-001 committed:
Merge branch 'branch-Level-8'

Kitty-001 committed:
A-JavaDoc

Kitty-001 committed:
Modify the code to align with coding standard

Kitty-001 committed:
Merge branch 'branch-A-Coding-Standard'

Kitty-001 committed:
Add a find option for ChatBot interaction

Kitty-001 committed:
Modify line break to align with coding standard

Kitty-001 committed:
Merge branch 'branch-A-Coding-Standard'

Kitty-001 committed:
Resolve conflicts

Kitty-001 committed:
Create more classes for better OOP practice, organize files into packages

Kitty-001 committed:
Merge branch 'add-gradle-support' of https://github.com/Kitty-001/ip into branch-A-Gradle

Kitty-001 committed:
Merge branch 'branch-A-Gradle'

Kitty-001 committed:
Add tests to Storage and TaskList

JJtan2002 reviewed (commented):
Looks good to me!

JJtan2002 commented on src/main/java/kitty/Parser.java:
```diff
@@ -0,0 +1,106 @@
+
+    public static boolean parseFirstWord(String str, Ui ui, TaskList tasks, Storage storage) throws FindException, MarksException, NumberFormatException, IndexOutOfBoundsException{
+        String[] aux = str.split(" ", 2);
+        switch (aux[0].trim()) {
+            case "find" -> parseFind(str, ui, tasks);
+            case "list" -> new ListCommand(ui, tasks).run();
+            case "mark", "delete", "unmark" -> parseMarks(str, ui, storage, tasks);
+            case "todo", "deadline", "event" -> new AddCommand(ui, tasks, str, storage).run();
+            default -> ui.showErrorMessage("Burrrrr~ What is this??? I have no idea about it...");
```
Kindly indent your switch case statements to follow the style guide:
```java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}

JJtan2002 reviewed (changes requested):
Just some minor changes to style please. Great job!

Kitty-001 committed:
Modify text-ui-test

Kitty-001 committed:
Merge branch 'A-TextUiTesting'

Kitty-001 committed:
Add style check

Kitty-001 committed:
Merge branch 'branch-A-CheckStyle'

Kitty-001 committed:
Adjust to comply with style check

Kitty-001 committed:
Adjust build.gradle to resolve error

Kitty-001 committed:
Set up JavaFX

Kitty-001 committed:
Design the layout

Kitty-001 committed:
Style the control

Kitty-001 committed:
Echo user input and auto scroll down

Kitty-001 committed:
Add dialogbox for response

Kitty-001 committed:
Refactor Kitty to use FXML

Kitty-001 committed:
Link Gui response with Kitty response

Kitty-001 committed:
Merge branch 'branch-JavaFX'

Kitty-001 committed:
Merge branch 'master' into branch-Level-10

Kitty-001 committed:
Merge branch 'branch-Level-10'

arnav-goel10 commented on src/main/java/kitty/Kitty.java:
```diff
@@ -0,0 +1,69 @@
+        ui.run();
+    }
+
+    /**
+     * Generates a response for the user's chat message.
+     */
```
The JavaDoc for this method should also describe the parameters and return value as stated in the coding standards.

![image](https://github.com/user-attachments/assets/f361b72c-1b7a-407c-bd0c-8ebb4213bb3b)

arnav-goel10 commented on src/main/java/kitty/TaskList.java:
```diff
@@ -0,0 +1,64 @@
+        int count = 1;
+        Task[] tmp = new Task[0];
+        StringBuilder message = new StringBuilder();
+        for (Task item: tasks.toArray(tmp)) {
```
Nice job using StringBuilder for efficient string concatenation. Simplifying the loop by removing the unnecessary array conversion will make the code even cleaner.

arnav-goel10 commented on src/main/java/kitty/Ui.java:
```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    public void showGreetingMessage() {
```
I noticed that showGreetingMessage() and showGoodbyeMessage() have similar code for displaying messages and division lines. You could simplify things by creating a common method for these repetitive parts. It could make your code cleaner and easier to manage!

Kitty-001 committed:
Modify build.gradle and create new jar file

Kitty-001 committed:
Add assertions to check input assumptions

By the time Parser.parseFirstWord process the user input, if it
continues to call other methods, the input to such functions should
fulfill some basic assumptions.

Add lines of assertions in files so as to check the effectiveness of
Parser and readiness of relevant inputs to some methods.

Kitty-001 committed:
Adjust code to enhance coding style

Kitty-001 committed:
Modify Storage.java and StorageTest.java

Originally Storage.java only work on a default file "data/Kitty.txt",
thus both the Chatbot and StorageTest work on this file. Modify the
Storage.java to ensure that it works on the file specified during the
initialization. This adds generality to the Storage class and
helps to separate the Chatbot and Test.

Kitty-001 committed:
Attempt to enhance code quality

Kitty-001 committed:
Merge pull request #1 from Kitty-001/branch-A-Assertions

Add assertions to check input assumptions

Kitty-001 committed:
Resolve merge conflict by incorporating both suggestions

Kitty-001 committed:
Create gradle.yml

Kitty-001 committed:
Adjust order

Kitty-001 committed:
Adjust .gitignore

Kitty-001 committed:
Adjust Storage.java for styleCheck

Kitty-001 committed:
Refactor AddCommand

Kitty-001 committed:
Refactor Parser

Kitty-001 committed:
Add tagging feature

This allows users to create tasks with or without tag. Users are also
allowed to update tag by the command "tag <index> <tag_name>".

Kitty-001 committed:
Adjust gui according to personality

Kitty-001 committed:
Update user guide and fix minor bugs

Kitty-001 committed:
Add theme to User Guide

Kitty-001 committed:
Adding reference to the content table

Kitty-001 committed:
Add file to github page

Kitty-001 committed:
Update README.md

Kitty-001 committed:
Adjust code to comply with styling

Kitty-001 committed:
Add more documentation

Kitty-001 committed:
Delete redundant file data/Test.txt
